src/async_fuse/memfs/s3_metadata.rs

```diff
@@ -859,49 +852,35 @@ impl<S: S3BackEnd + Sync + Send + 'static, K: KVEngine + 'static> MetaData for S
         flags: u32,
     ) -> DatenLordResult<usize> {
         let data_len = data.len();
-        let (result, full_path, parent_ino) = {
-            let mut cache = self.cache().write().await;
-            let inode = cache.get_mut(&ino).unwrap_or_else(|| {
+        let (result, full_path, _) = {
+            let mut inode = self.get_node_from_kv_engine(ino).await.unwrap_or_else(|| {
                 panic!(
                     "write() found fs is inconsistent, \
                      the inode ino={ino} is not in cache"
                 );
             });
-            let parent_ino = inode.get_parent_ino();
+            let parent_ino = inode.as_ref().get_parent_ino();
 
             debug!(
                 "write_helper() about to write {} byte data to file of ino={} \
                 and name {:?} at offset={}",
                 data.len(),
                 ino,
-                inode.get_name(),
+                inode.as_ref().get_name(),
                 offset
             );
             let o_flags = fs_util::parse_oflag(flags);
             let write_to_disk = true;
             let res = inode
+                .as_mut_ref()
                 .write_file(fh, offset, data, o_flags, write_to_disk)
                 .await;
-            (res, inode.get_full_path().to_owned(), parent_ino)
+            (res, inode.as_ref().get_full_path().to_owned(), parent_ino)
         };
         self.invalidate_remote(&full_path, offset, data_len).await;
-        self.sync_attr_remote(&full_path).await;
-        // update dir
-        let cache = self.cache().write().await;
-        let parent = cache.get(&parent_ino).unwrap_or_else(|| {
-            panic!(
-                "write() found fs is inconsistent, \
-                        the inode ino={ino} is not in cache"
-            );
-        });
-        self.persist_handle.mark_dirty(
-            parent.get_ino(),
-            PersistDirContent::new_from_cache(
-                parent.full_path().to_owned(),
-                parent.get_dir_data(),
-                serial::file_attr_to_serial(&parent.get_attr()),
-            ),
-        );
+        // We don't need to sync the dirty node because it will be synced when it is dropped
+        // We don't need to update parent dir here
+        // Because when the parent dir is loaded from KV, it will be updated
```

Comment:
We don't need explain here as the design is totally different. The comments is used to explain the change from the old version to the new one, but code readers won't know what's the old code if they don't review the logs.

If you really want to explain the change, please explain it in the commit message, which is more clear for a change history viewer.